src/GraphControl/Control/Grapher.cpp

```diff
@@ -1088,10 +1099,21 @@ optional<vector<shared_ptr<Graphing::IEquation>>> Grapher::TryInitializeGraph(bo
 {
     if (keepCurrentView || IsKeepCurrentView)
     {
+        auto renderer = m_graph->GetRenderer();
         double xMin, xMax, yMin, yMax;
-        m_graph->GetRenderer()->GetDisplayRanges(xMin, xMax, yMin, yMax);
+        renderer->GetDisplayRanges(xMin, xMax, yMin, yMax);
         auto initResult = m_graph->TryInitialize(graphingExp);
-        m_graph->GetRenderer()->SetDisplayRanges(xMin, xMax, yMin, yMax);
+        if (IsKeepCurrentView)
+        {
+            renderer->PrepareGraph();
```

Comment:
Not sure if related to this change or internal graph engine changes, but when switching to gradians with y=x, sin(x), and x^2+y^2=1 graphed I see a new crash

Reply:
I can't repro this crash.

Reply:
Okay, I checked master and it repros there too so likely not an issue with this PR.